sessionctx/variable/session.go

```diff
@@ -1294,6 +1298,16 @@ func (s *SessionVars) SetSystemVar(name string, val string) error {
 		s.FoundInPlanCache = TiDBOptOn(val)
 	case TiDBEnableCollectExecutionInfo:
 		config.GetGlobalConfig().EnableCollectExecutionInfo = TiDBOptOn(val)
+	case SQLSelectLimit:
+		if strings.EqualFold(val, "default") {
+			s.SelectLimit = math.MaxUint64
+		} else {
+			result, err := strconv.ParseUint(val, 10, 64)
+			if err != nil {
+				return errors.Trace(err)
+			}
+			s.SelectLimit = result
```

Comment:
Maybe `variable.SQLSelectLimit` needs to be put into [`builtinGlobalVariable`](https://github.com/pingcap/tidb/blob/7ca3d9cadccaa19b12e20c2b5d71615c7e735a5d/session/session.go#L1973)